diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/LineLength.kt

```diff
@@ -74,6 +119,200 @@ class LineLength : Rule("line-length") {
+    }
+
+    private fun fixComment(wrongNode: ASTNode, configuration: LineLengthConfiguration){
+        var node = wrongNode
+        do {
+            val commentText = node.text
+            var indexLastSpace = configuration.lineLength.toInt()
+            while (commentText[indexLastSpace] != ' ') {
+                --indexLastSpace
+                if (indexLastSpace == 0)
+                    return
+            }
+            val nodeText = "//${commentText.substring(indexLastSpace, commentText.length)}"
+            val newNode = LeafPsiElement(EOL_COMMENT, nodeText)
+            node.treeParent.run {
+                addChild(LeafPsiElement(EOL_COMMENT, commentText.substring(0, indexLastSpace)), node)
+                addChild(PsiWhiteSpaceImpl("\n"), node)
+                addChild(newNode, node)
+                removeChild(node)
+            }
+            node = newNode
+        } while (node.text.length > configuration.lineLength)
+    }
+
+    private fun fixCondition(wrongNode: ASTNode, configuration: LineLengthConfiguration,
+                             leftInitOffset: Int = -1, isProperty: Boolean = false) {
+        var leftOffset = if (leftInitOffset < 0){
+            positionByOffset(wrongNode.firstChildNode.startOffset).second
+        } else leftInitOffset
+        val binList = mutableListOf<ASTNode>()
+        if (isProperty)
+            dfsForProperty(wrongNode, binList)
+        else
+            searchBinaryExpression(wrongNode, binList)
+        lateinit var binaryText: String
+        binList.forEachIndexed { index, astNode ->
+            if (index == 0){
+                binaryText = astNode.text
+            } else {
+                val operationRef = astNode.treeParent.findChildByType(OPERATION_REFERENCE)!!
+                if (operationRef.treePrev.elementType == WHITE_SPACE)
+                    binaryText += operationRef.treePrev.text
+                binaryText += operationRef.text
+                if (operationRef.treeNext.elementType == WHITE_SPACE)
+                    binaryText += operationRef.treeNext.text
+                binaryText+= astNode.text
+                if (leftOffset + binaryText.length > configuration.lineLength) {
+                    val commonParent = findCommonParent(astNode, binList[index-1])
+                    val newLine = PsiWhiteSpaceImpl("\n")
+                    commonParent.addChild(newLine, commonParent.findChildByType(OPERATION_REFERENCE)!!.treeNext)
+                    leftOffset = 0
+                    binaryText = astNode.text
+                }
+            }
+        }
+    }
+
+    private fun findCommonParent(firstNode: ASTNode ,secondNode: ASTNode): ASTNode{
+        var firstParent: ASTNode? = firstNode
+        while (firstParent!= null) {
+            if (depthFirstSearch(firstParent, secondNode)!= null)
+                return firstParent
+            firstParent = firstParent.treeParent
+        }
+        return firstParent ?: firstNode
```

Comment:
Why do you return `firstNode` if `firstParent` is `null`? If we have reached this line and `firstParent` is `null`, this should mean that the only common parent is `FILE` node, and you are returning `firstNode` instead.